checker.c

```diff
@@ -1,21 +1,54 @@
-#include <stdio.h>
-#include <assert.h>
 
-int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
+#include "checker.h"
+
+int IsTempatureOK(float temprature){
+  checkValueRange(temprature, TempMAX, TempMIN);
+}
+
+int isStateOfChargeOK(float soc)
+{
+  checkValueRange(soc, SOCMAX, SOCMIN);
+}
+
+int isChargeRateOK(float chargeRate)
+{
+  checkChargeRateLimit( chargeRate);
+}
+
+int checkValueRange(float value, float MAX, float MIN)
+{
+  if (value < MIN || value > MAX)
+  {
     printf("Temperature out of range!\n");
-    return 0;
-  } else if(soc < 20 || soc > 80) {
-    printf("State of Charge out of range!\n");
-    return 0;
-  } else if(chargeRate > 0.8) {
+    return FALSE;
+  }
+  else
+    return TRUE;
+}
+
+int checkChargeRateLimit(float value)
+{
+  if (value > ChangeRate)
+  {
     printf("Charge Rate out of range!\n");
-    return 0;
+    return FALSE;
   }
-  return 1;
+  else return TRUE;
 }
 
-int main() {
+int batteryIsOk(float temperature, float soc, float chargeRate)
+{
+return  ( IsTempatureOK(temperature)) && ( isStateOfChargeOK(soc)) && (isChargeRateOK(chargeRate));
+}
+
+void Tester()
+{
   assert(batteryIsOk(25, 70, 0.7));
-  assert(!batteryIsOk(50, 85, 0));
+  assert((batteryIsOk(50, 85, 0))==FALSE);
```

Comment:
You'll need more asserts to cover all conditions. You could also test all conditions in `checkValueRange` and reduce the amount of variations you need to test

Reply:
yes i have updated the tester will couple of more cases